woocommerce/wc-admin
# Refactor D3Chart

Fixes #1492.
Fixes #1514.

This PR refactors `<D3Chart />` with a couple of goals:
- Split the `params` variable into smaller ones (`formats`, `params` and `scales`).
- Move variables as close as possible to the place they are used. For example `dateSpaces` is only used in line charts, so it makes more sense to create it in `drawLines()` than doing it in `<D3Chart />`.

### Accessibility


### Detailed test instructions:
For #1492:
- Go to the _Products_ report, for example.
- In the date picker, select a single day that you know has orders.
- In the chart interval, select _By day_.
- Verify the dots are centered.
![image](https://user-images.githubusercontent.com/3616980/52655734-2985a380-2ef5-11e9-9b84-185dfb0eaed2.png)

For #1514:
- Go to the _Products_ report in IE 11.
- Under the _Show_ filter, select _Product Comparison_.
- Select 2 or more products.
- In the date picker, select a wide range.
- Select the _bar chart_.
- Verify the X-axis labels are visible.
![image](https://user-images.githubusercontent.com/3616980/52655993-b4669e00-2ef5-11e9-9444-5b6c3757a287.png)

In addition to that, playing a bit with the chart: switching between line and bar types, trying different date ranges/intervals, resizing the viewport and hovering the lines/bars would help in order to detect any regression.
